### PR TITLE
[SR-4375] Support more multi unescape delimiter

### PR DESCRIPTION
**Pull request by Astralidea**
The previous usage is not supported \n, \t, \r directly.
If you want to use \n, \r, \t，you should use its ascii HEX character instead.
Like the following example:
```
"column_separator:\x0A" to replace the \n
"column_separator:\x0D" to replace the \r
"column_separator:\x09" to replace the \t
``` 

This PR is to support unescape character like \n, \r, \t.
For 0x case if cannot convert to one ascii character it will use multi original character instead.
The previous usage is not supported is \x0001, to use \x00\x01 instead.

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java [hunk 1]
```diff
@@ -31,29 +31,86 @@ public static String convertDelimiter(String originStr) throws AnalysisException
...
+
+        return writer.toString();
+    }
 
-            // transform to delimiter
-            StringWriter writer = new StringWriter();
-            for (byte b : hexStrToBytes(hexStr)) {
-                writer.append((char) b);
+    private static int parseHexString(String originStr, StringWriter writer, int offset) throws AnalysisException {
+        if (offset+4 > originStr.length()) {
```
Comment (DeepThinker666) [inline fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java:93]: offset + 4 > originStr.length()

File: fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java [hunk 2]
```diff
@@ -31,29 +31,86 @@ public static String convertDelimiter(String originStr) throws AnalysisException
...
+            writer.append(originStr, offset, offset + originStr.length());
+            return originStr.length();
+        }
+        String hexStr = originStr.substring(offset+2, offset+4);
```
Comment (DeepThinker666) [inline fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java:97]: originStr.substring(offset + 2, offset + 4);

File: fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java [hunk 3]
```diff
@@ -31,29 +31,86 @@ public static String convertDelimiter(String originStr) throws AnalysisException
...
+        }
+
+        return writer.toString();
+    }
 
-            // transform to delimiter
-            StringWriter writer = new StringWriter();
-            for (byte b : hexStrToBytes(hexStr)) {
-                writer.append((char) b);
+    private static int parseHexString(String originStr, StringWriter writer, int offset) throws AnalysisException {
```
Comment (DeepThinker666) [inline fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java:92]: you can add some cases here as example

**Review by DeepThinker666 — CHANGES_REQUESTED** [review]
Is the delimiter for load?
could you give some examples?

**Comment by Astralidea** [comment]
> Is the delimiter for load?
> could you give some examples?

yes, the example in ColumnSeparatorTest.java

File: fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java [hunk 4]
```diff
@@ -31,29 +31,97 @@ public static String convertDelimiter(String originStr) throws AnalysisException
...
-                throw new AnalysisException("Invalid delimiter '" + originStr + ": hex length must be a even number");
-            }
-            for (char hexChar : hexStr.toUpperCase().toCharArray()) {
-                if (HEX_STRING.indexOf(hexChar) == -1) {
-                    throw new AnalysisException("Invalid delimiter '" + originStr + "': invalid hex format");
-                }
+        StringWriter writer = new StringWriter();
+
+        for (int i = 0; i < originStr.length(); i++) {
+            char ch = originStr.charAt(i);
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java:37]: what's the ch is short for ?

File: fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java [hunk 5]
```diff
@@ -31,29 +31,97 @@ public static String convertDelimiter(String originStr) throws AnalysisException
...
+            boolean outputOneChar = true;
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java:38]: What's the purpose of this pull request? It's to support \n to be column separator>

File: fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java (hunk 4 above)
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java:37]: character

File: fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java [hunk 6]
```diff
@@ -31,29 +31,97 @@ public static String convertDelimiter(String originStr) throws AnalysisException
...
+        \r Insert a carriage
+        \f Insert a formed
+        \' Insert a single quote character
+        \" Insert a double quote character
+        \\ Insert a backslash
+        \x Insert a hex escape e.g. \x48 represent h
+     */
+    private static int parseHexString(String originStr, StringWriter writer, int offset) throws AnalysisException {
+        if (offset + 4 > originStr.length()) {
+            writer.append(originStr, offset, offset + originStr.length());
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java:105]: writer.append(originStr, offset, originStr.length());

File: fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java [hunk 7]
```diff
@@ -31,29 +31,97 @@ public static String convertDelimiter(String originStr) throws AnalysisException
...
+            return originStr.length();
+        }
+        String hexStr = originStr.substring(offset + 2, offset + 4);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java:108]: hexStr is definitely two characters, why need to check empty or even length？

File: fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java (hunk 5 above)
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java:38]: yes

File: fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java (hunk 7 above)
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java:108]: This is the logic from before, I kept it, and now I think I cant delete it right now

**Review by chaoyli — APPROVED** [review]

**Pull request merged**